microsoft/testsuites/nvme/nvme.py

```diff
@@ -328,27 +329,35 @@ def nvme_manage_ns_validation(self, node: Node) -> None:
         #  'nvme id-ctrl device', nvme namespace can be created, deleted and detached,
         #  otherwise it can't be managed.
         if not nvme_cli.support_ns_manage_attach(nvme_devices[0]):
-            # NVMe Status:INVALID_OPCODE(1)
-            ns_management_exit_code = 1
+            # for old nvme cli version, it returns 22
+            # for new nvme cli version, it returns 1
+            # refer https://github.com/linux-nvme/nvme-cli/issues/1120
+            ns_management_exit_code = [1, 22]
         for namespace in nvme_namespaces:
             # 2. `nvme format namespace` - format a namespace.
             format_namespace = nvme_cli.format_namespace(namespace)
             format_namespace.assert_exit_code(device_format_exit_code)
             # 3. `nvme create-ns namespace` - create a namespace.
             create_namespace = nvme_cli.create_namespace(namespace)
-            # NVMe Status:INVALID_OPCODE: The associated command opcode field is not
-            #  valid(1) => exit code 1
-            # NVMe status: INVALID_OPCODE: The associated command opcode field is not
-            #  valid(0x1) => exit code 22
-            if "(0x1)" in create_namespace.stdout:
-                ns_management_exit_code = 22
-            create_namespace.assert_exit_code(ns_management_exit_code)
+            if create_namespace.exit_code not in ns_management_exit_code:
```

Comment:
When the logic goes to complex, it's time to refactor. For this scenario, please change assert to supports a list of exit code.

Reply:
assert already supports list of exit code